api/solarperformanceinsight_api/compute.py

```diff
@@ -30,22 +35,326 @@ def run_job(job_id: UUID, user: str):
+        poa_global, effective_irradiance, and cell_temperature over all arrays
+    """
+    # run chain
+    mc = getattr(chain, run_model_method)(weather_data)
+    results = mc.results
+    adjust = partial(_adjust_frame, tshift=tshift)
+    performance = adjust(results.ac, "performance")
+    summary_frame = pd.DataFrame(
+        {
+            "poa_global": 0,  # type: ignore
+            "effective_irradiance": 0,  # type: ignore
+            "cell_temperature": 0,  # type: ignore
+        },
+        index=performance.index,
+    )
+    summary_frame.index.name = "time"  # type: ignore
+    # make nice result output
+    num_arrays = len(mc.system.arrays)
+    out = []
+    for i in range(num_arrays):
+        weather: pd.DataFrame = adjust(  # type: ignore
+            _get_index(results, "total_irrad", i)[["poa_global"]], "poa_global"
+        )
+        weather.loc[:, "effective_irradiance"] = adjust(  # type: ignore
+            _get_index(results, "effective_irradiance", i), "effective_irradiance"
+        )
+        weather.loc[:, "cell_temperature"] = adjust(  # type: ignore
+            _get_index(results, "cell_temperature", i), "cell_temperature"
+        )
+        summary_frame += weather  # type: ignore
+        out.append(
+            DBResult(
+                schema_path=f"/inverters/{inverter_num}/arrays/{i}",
+                type="weather data",
+                data=weather,
+            )
+        )
+    out.append(
+        DBResult(
+            schema_path=f"/inverters/{inverter_num}",
+            type="performance data",
+            data=performance,
+        )
+    )
+    # mean
+    summary_frame /= num_arrays  # type: ignore
+    summary_frame.insert(0, "performance", performance)  # type: ignore
+    summary_frame.insert(
+        len(summary_frame.columns),
+        "zenith",
+        adjust(results.solar_position["zenith"], "zenith"),  # type: ignore
+    )  # type: ignore
+    return out, summary_frame
+
+
+def run_performance_job(job: models.StoredJob, si: storage.StorageInterface):
+    """Compute the performance, and other modeling variables, for the Job and
+    store the inverter level performance, total system performance, array level weather,
+    and a monthly summary to the database for retrieval
+    """
+    job_time_range = job.definition.parameters.time_parameters._time_range
+    summary = pd.DataFrame(
+        {
+            "performance": 0,  # type: ignore
+            "poa_global": 0,  # type: ignore
+            "effective_irradiance": 0,  # type: ignore
+            "cell_temperature": 0,  # type: ignore
+            "zenith": 0,  # type: ignore
+        },
+        index=job_time_range,
+    )
+    summary.index.name = "time"  # type: ignore
+    run_model_method = job.definition._model_chain_method
+    tshift = job.definition.parameters.time_parameters.step / 2
+    chains = construct_modelchains(job.definition.system_definition)
+
+    weather_count = 0.0
```

Comment:
why is this a float?

Reply:
old habit to make sure result after division is float